uboot-mvebu: remove enabled CONFIG_CMD_SETEXPR

BKPepe opened the pull request:
This U-boot option is already enabled for Turris Omnia in defconfig.
Other devices like Helios4 and Clearfog has this option as not set. If
it is used somehow, then it should be corrected in defconfigs for
related boards.

It is useful when passing initramfs end address.

It was introduced by this commit https://github.com/openwrt/openwrt/commit/0a3e07b2f5a5a8b20da173663007cb1f4b8f3eb8 (cc: @artox) , unfortunately, the commit itself is not explaining why it needs to be enabled for all boards and why it is necessary.

BKPepe committed:
uboot-mvebu: remove enabled CONFIG_CMD_SETEXPR

We don't need to make sure that we want to have enabled
CONFIG_CMD_SETEXPR by default, since this is already done in U-boot [1].
This was actually needed only for clearfog board [2], which was added in
commit: da0005a6d08ae33d958a6d8a6c0c12dc07b5b2b8 ("uboot-mvebu: add
patch to enable setexpr for clearfog boards) and send to U-boot to fix
it properly. After a while, there was added support for Turris Omnia,
which uses setexpr as well [3], but for this board, there are no fixes
needed in U-boot and that's why we can remove this option here.

It is helpful with shell scripting. If some downstream distributions are
using it, they should correct it in defconfig for related boards.

[1] https://source.denx.de/u-boot/u-boot/-/blob/e95afa56753cebcd20a5114b6d121f281b789006/cmd/Kconfig#L1504

[2] https://github.com/openwrt/openwrt/blob/852126680e21edc71c0c66561ae5a6d7479dcc67/target/linux/mvebu/image/clearfog.bootscript#L7

[3] https://github.com/openwrt/openwrt/blob/852126680e21edc71c0c66561ae5a6d7479dcc67/target/linux/mvebu/image/turris-omnia.bootscript#L2

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>

Artox commented:
`setexpr` is used to to " calculate"  the rootfs partition number - being boot partition + 1 - to then lookup the uuid of that partition, to be used for the `root=` kernel cmdline: https://github.com/openwrt/openwrt/blob/master/target/linux/mvebu/image/clearfog.bootscript#L6

Imo if setexp is needed by the distro (here openwrt), then openwrt should enable it, not the defconfig in u-boot.
Other projects using the Clearfog U-Boot port such as Debian e.g. do not use `setexpr` in their boot-script.

BKPepe commented:
Actually, in OP, I was wrong. 

It seems you wanted to have it enabled only for clearfog, and it was introduced by this commit https://github.com/openwrt/openwrt/commit/e2814f641600efabfa906fa0a5957eaf055a67e5 . Then once time pass, it is **enabled to every single board for mvebu**, and you linked clearfog. This single board is in U-boot defconfig has ``# CONFIG_CMD_SETEXPR is not set``

Ref: https://source.denx.de/u-boot/u-boot/-/blob/e95afa56753cebcd20a5114b6d121f281b789006/configs/clearfog_defconfig#L37

This is not what Turris Omnia [has](https://source.denx.de/u-boot/u-boot/-/blob/e95afa56753cebcd20a5114b6d121f281b789006/configs/turris_omnia_defconfig), which is using setexpr here as well. If you follow U-boot instructions to compile it, you will see that it is enabled for Turris Omnia:

1. Clone U-boot
2. Run: ``make turris_omnia_defconfig``
3. See ``.config``
Enabled under shell scripting command.

Other boards here in mvebu does not use setexpr in bootscript. You are still trying to do stuff, which should be done upstream not here. Because setexpr should already default: 
https://source.denx.de/u-boot/u-boot/-/blob/e95afa56753cebcd20a5114b6d121f281b789006/cmd/Kconfig#L1504

Artox commented:
setexpr enabled by default is an interesting find! Imo enabling or disabling setexpr was a distro-choice when I did, since not every distro needs it .... .

So I am not opposed to getting upstream to enable setexpr for the Clearfog.

Artox commented:
@BKPepe Thank you :)

BKPepe committed:
uboot-mvebu: add patch to enable setexpr for clearfog boards

Option CMD_SETEXPR is already default in U-boot [1], since this was
disabled since initial version for this board, there is send this
	patch to U-boot mailing list to enable it.

It is required to use in OpenWrt bootscript for these boards [2].

[1] https://source.denx.de/u-boot/u-boot/-/blob/e95afa56753cebcd20a5114b6d121f281b789006/cmd/Kconfig#L1504

[2] https://github.com/openwrt/openwrt/blob/852126680e21edc71c0c66561ae5a6d7479dcc67/target/linux/mvebu/image/clearfog.bootscript#L7

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>

hauke commented on package/boot/uboot-mvebu/Makefile:
```diff
@@ -63,9 +63,6 @@ UBOOT_TARGETS:= \
-	echo CONFIG_CMD_SETEXPR=y >> $(PKG_BUILD_DIR)/configs/$(UBOOT_CONFIG)_defconfig
-
 	$(call Build/Configure/U-Boot)
 endef
```
The `include/u-boot.mk` defines `Build/Configure/Default = $$$$(Build/Configure/U-Boot)`. Please remove the `Build/Configure` block here it is not needed any more.

BKPepe replied:
Ok, done.

hauke commented:
Thank you for your patches, I applied them to master in b3c2072504867b8ca00dde37f531edf0cfbe3359 and 69cef74c671cb7fa738a5a4c9a283c29f45326c7.
